### PR TITLE
Add missing commonMain folder to fuel-kotlinx-serialization project

### PR DESCRIPTION
**Pull request by iNoles**
### Description

This pull request adds the `commonMain` folder to the `fuel-kotlinx-serialization` project. The `commonMain` folder is necessary for the Kotlin Multiplatform setup and was missing from the project.

### Changes Made

- Added the `src/commonMain` directory.
- Defined an `expect` function `HttpResponse.toJson`
- Provided `actual` implementations for every platforms

### Motivation and Context

This change is required to properly organize the common code shared across different Kotlin Multiplatform platform platforms. Without this folder, the project setup is incomplete and may lead to build issues.

### CODE REVIEW

File: fuel-kotlinx-serialization/src/appleMain/kotlin/fuel/serialization/ResponseExtensions.kt [hunk 1]
```diff
@@ -6,8 +6,8 @@ import fuel.HttpResponse
 import kotlinx.serialization.DeserializationStrategy
 import kotlinx.serialization.json.Json
 
-public fun <T : Any> HttpResponse.toJson(
-    json: Json = Json { allowStructuredMapKeys = true },
+public actual fun <T : Any> HttpResponse.toJson(
+    json: Json,
```
Comment (kittinunf) [inline fuel-kotlinx-serialization/src/appleMain/kotlin/fuel/serialization/ResponseExtensions.kt:10]: [q] actual method doesn't support the default parameter?

Comment (iNoles) [reply fuel-kotlinx-serialization/src/appleMain/kotlin/fuel/serialization/ResponseExtensions.kt:10]: It triggered the Kotlin IR flaws for expect/actual.

Comment (kittinunf) [reply fuel-kotlinx-serialization/src/appleMain/kotlin/fuel/serialization/ResponseExtensions.kt:10]: Okay!

**Review by kittinunf — APPROVED** [review]
❇️ Looks awesome

**Pull request merged**